src/history/historyTracker.ts

```diff
@@ -427,14 +428,16 @@ export class HistoryTracker {
    * used to look like.
    */
   public addChange(cursorPosition = [new Position(0, 0)]): void {
-    const newText = this._getDocumentText();
+    // Get the current document's text which is conveniently equals to the zero based offset at the document's end.
+    const documentEnd = cursorPosition[0].getDocumentEnd();
+    const newTextLength = this.vimState.editor.document.offsetAt(documentEnd);
```

Comment:
Perhaps we can just enumerate the special cases and deal with them separately, but doesn't this break with `:s/abc/def`, for instance?